### DISCUSSION TITLE
Trino memory usage increases with enabling jmx for metrics collection

### DISCUSSION THREAD

**Discussion by hsushmitha** [trinodb/trino#24198] [General]
We recently enabled jmx for metrics collection in Trino 463 version. we see almost 21% increase in memory usage with just system flush metadata calls happening in Trino. without jmx enabled Trino memory usage was 1% with flush metadata calls. we see significant increase in memory usage with jmx enabled. Is it expected that trino memory usage increases with jmx metrics collection? Are there alternatives to jmx metrcis collections. we would want to eventually collect trino metrics and use those values to trigger autoscaling.

**Comment by lozbrown** [comment]
Are you referring to the JMX agent that has been integrated in the helm chart?

If so Trino natively supports OpenMetrics and these can be collected instead, unfortunatly Documentation for OpenMetrics is outstanding. 

FYI @mosabua

**Comment by hsushmitha** [comment]
Are we referring to open telemetry mentioned here: https://trino.io/docs/current/admin/opentelemetry.html.

**Comment by hashhar** [comment]
if you are able to reproduce this would you be willing to take a heap histogram before and after the memory growth?
